uswds/uswds-compile
# Update dependencies & add Snyk

This PR:
- Security fixes via `overrides`
- Updates dependencies
- Adds Snyk monitor via CLI

Closes #48.

## Dependency updates

| Dependency                     |     Old | 25da089 [04/27/23] | b37aee6 [07/12/23] |
| :----------------------------- | ------: | -----------------: | -----------------: |
| `@uswds/uswds`                 |   3.3.0 |              3.4.1 |              3.5.0 |
| `uswds`                        |  2.13.3 |             2.14.0 |                  - |
| `autoprefixer`                 | 10.4.13 |            10.4.14 |                  - |
| `del`                          |   6.0.0 |              6.1.1 |                  - |
| `glob-parent`  [⚠️ via overrides] |       - |              6.0.2 |                  - |
| `gulp-replace`                 |   1.1.3 |              1.1.4 |                  - |
| `postcss`                      |  8.4.19 |             8.4.23 |             8.4.25 |
| `sass-embedded`                |  1.56.1 |             1.62.0 |             1.63.6 |
| `unset-value` [⚠️ via overrides]  |       - |              2.0.1 |                  - |

> **Note**
> There is an additional update for `del@7`, but that **only** supports ESM.

## How to test

<details><summary>Confirm errors</summary>
<p></p>

1. Checkout `git checkout develop` branch
1. Run `npm i && npm audit`
1. Confirm error message: `6 high severity vulnerabilities`
1. Run `npx snyk test`
1. Confirm error message: `found 2 issues, 33 vulnerable paths.`
1. Reset branch with `git checkout -- .`

</details> 
 
<details><summary>Confirm fix</summary>

<p></p>

1. Checkout branch `git checkout  jm-update-dependencies`
1. Run `npm i && npm audit`
1. Confirm 0 vulnerabilities
1. Run snyk test `npx snyk test`
1. Confirm 0 vulnerabilities.

> **Note**
> I tried testing this on uswds-sandbox with mixed results. Installing via 
>npm link will show dependency fixes, but gulp scripts won't run (even if Node versions match). Installing via git branch won't show security fixes, but scripts **will** run.
</details>  



Test functionality:

- All gulp tasks should run without errors. Test branch available in Sandbox [branch `test-compile-60`](https://github.com/uswds/uswds-sandbox/tree/test-compile-60) or use site.
- There should be zero vulnerabilities in `npm audit`.
